src/main/java/org/quiltmc/config/api/annotations/ChangeWarning.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.config.api.annotations;
+
+import org.quiltmc.config.api.Config;
+import org.quiltmc.config.api.metadata.MetadataType;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Optional;
+
+/**
+ * Used to tell config screen libraries that a warning should be displayed before applying changes. Can be applied to configs, sections and properties.
```

Comment:
```suggestion
 * Used to tell config screen libraries that a warning should be displayed before applying changes. Can be applied to configs, sections and fields.
```
properties isn't the word I use on the wiki and I think the average person understands "config field" better than "config property"